rpcs3/rpcs3qt/skylander_dialog.cpp

```diff
@@ -0,0 +1,537 @@
+		// Type 2
+		res_crc                         = do_crc_blocks(0xFFFF, {1, 2, 4});
+		*((u16*)&decrypted_header[0xC]) = res_crc;
+
+		// Type 1
+		*((u16*)&decrypted_header[0xE]) = 5;
+		*((u16*)&decrypted_header[0xE]) = skylander_crc16(0xFFFF, decrypted_header.data(), 16);
+
+		set_block(active, decrypted_header);
+		set_block(active + 9, sub_header);
+	}
+}
+
+void skylander_dialog::new_skylander()
+{
+	QString filePath = QFileDialog::getSaveFileName(this, tr("Create Skylander File"), cur_sky_file_path, tr("Skylander Object (*.sky);;"));
+	if (filePath == NULL)
+		return;
+
+	if (sky_file)
+		sky_file.close();
+
+	sky_file.open(filePath.toStdString(), fs::read + fs::write + fs::create);
+	if (!sky_file)
+		return;
+
+	cur_sky_file_path = filePath;
+
+	{
+		std::lock_guard lock(sky_mutex);
+		memset(sky_dump, 0, 0x40 * 0x10);
+
+		// Set the block permissions
+		*((u32*)&sky_dump[0x36]) = 0x690F0F0F;
+		for (u32 index = 1; index < 0x10; index++)
+		{
+			*((u32*)&sky_dump[(index * 0x40) + 0x36]) = 0x69080F7F;
+		}
+
+		*((u16*)&sky_dump[0x1E]) = skylander_crc16(0xFFFF, sky_dump, 0x1E);
+
+		std::array<u8, 16> zero_array = {};
+		for (u32 index = 8; index < 0x40; index++)
+		{
+			if ((index + 1) % 4)
+			{
+				set_block(index, zero_array);
+			}
+		}
+
+		set_checksums();
+
+		sky_reload = true;
+	}
+
+	sky_save();
+
+	edit_curfile->setText(filePath);
+	update_edits();
+}
+
+void skylander_dialog::load_skylander()
+{
+	QString filePath = QFileDialog::getOpenFileName(this, tr("Select Skylander File"), cur_sky_file_path, tr("Skylander (*.sky);;"));
+	if (filePath == NULL)
+		return;
+
+	if (sky_file)
+		sky_file.close();
+
+	sky_file.open(filePath.toStdString(), fs::read + fs::write);
+	if (!sky_file)
+		return;
+
+	cur_sky_file_path = filePath;
+
+	sky_load();
+
+	edit_curfile->setText(filePath);
+	update_edits();
+}
+
+void skylander_dialog::update_edits()
+{
+	auto widget_enabler = [&](bool status) {
+		combo_skylist->setEnabled(status);
+		edit_skyID->setEnabled(status);
+		edit_skyXP->setEnabled(status);
+		edit_skyMoney->setEnabled(status);
+		button_update->setEnabled(status);
+	};
+
+	if (!sky_file)
+	{
+		widget_enabler(false);
+		return;
+	}
+	else
+	{
+		widget_enabler(true);
+	}
+
+	{
+		std::lock_guard lock(sky_mutex);
+		edit_skyID->setText(QString::number(*((u16*)&sky_dump[0x10])));
+
+		u8 active = get_active_block();
+
+		std::array<u8, 16> decrypted;
+		get_block(active, decrypted);
+		u32 xp = (*((u32*)decrypted.data())) & 0xFFFFFF;
+		edit_skyXP->setText(QString::number(xp));
+		u16 money = *((u16*)&decrypted[3]);
+		edit_skyMoney->setText(QString::number(money));
+	}
+}
+
+void skylander_dialog::process_edits()
+{
+	{
+		std::lock_guard lock(sky_mutex);
+
+		bool cast_success = false;
+		u16 skyID         = edit_skyID->text().toInt(&cast_success);
+		if (cast_success)
+		{
+			*((u16*)&sky_dump[0x10]) = skyID;
+			*((u32*)&sky_dump[0])    = skyID;
+		}
+
+		*((u16*)&sky_dump[0x1E]) = skylander_crc16(0xFFFF, sky_dump, 0x1E);
```

Comment:
Don't hadcode this type's endianess, use le_t<u16> instead. (and in other cases as well)

Reply:
I mean, the actual action result is dependant on the host endisness here.